will/backends/io_adapters/slack.py

```diff
@@ -457,85 +619,72 @@ def _update_channels(self):
-                name=v.real_name,
-            )
-            if v.name == self.handle:
-                self.me = Person(
-                    id=v.id,
-                    mention_handle="<@%s>" % v.id,
-                    handle=v.name,
-                    source=clean_for_pickling(v),
-                    name=v.real_name,
-                )
-            if user_timezone and user_timezone != 'unknown':
-                people[k].timezone = user_timezone
-                if v.name == self.handle:
-                    self.me.timezone = user_timezone
+        if client:
+            for page in client.users_list(limit=self.PAGE_LIMIT):
+                for member in page["members"]:
+                    if member["deleted"]:
+                        continue
+                    member_id = member["id"]
+                    user_timezone = member.get("tz")
+                    people[member_id] = Person(
+                        id=member_id,
+                        mention_handle=member.get("mention_handle", "SECRETAGENTMAN"),
+                        handle=member["name"],
+                        source=clean_for_pickling(member),
+                        name=member.get("real_name", "SECRETAGENTMAN"),
+                    )
+                    if member["name"] == self.handle:
+                        self.me = people[member_id]
+                    if user_timezone and user_timezone != "unknown":
+                        people[member_id].timezone = user_timezone
         if len(people.keys()) == 0:
             # Server isn't set up yet, and we're likely in a processing thread,
             if self.load("slack_people_cache", None):
                 self._people = self.load("slack_people_cache", None)
-            if not hasattr(self, "me") or not self.me:
+            if self.me is None:
                 self.me = self.load("slack_me_cache", None)
-            if not hasattr(self, "handle") or not self.handle:
+            if self.handle is None:
                 self.handle = self.load("slack_handle_cache", None)
         else:
             self._people = people
             self.save("slack_people_cache", people)
             self.save("slack_me_cache", self.me)
             self.save("slack_handle_cache", self.handle)
 
-    def _update_backend_metadata(self):
-        self._update_people()
-        self._update_channels()
+    def _update_backend_metadata(self, **event):
+        "Updates all our internal caches.  Very expenseive"
+        logging.debug("updating backend on event: %s", event)
+        name = event.get("data", dict()).get("self", dict()).get("name")
+        if name is not None:
+            self.__class__.handle = name
+        Thread(
+            target=self._update_people, args=(event["web_client"],), daemon=True
+        ).start()
+        Thread(
+            target=self._update_channels, args=(event["web_client"],), daemon=True
+        ).start()
 
     def _watch_slack_rtm(self):
```

Comment:
It looks like this should resolve any issues where an unhandled exception aborts the loop.